odoo/fields.py

```diff
@@ -2792,13 +2795,77 @@ def _record_ids_per_res_model(self, records):
         return model_ids
 
 
+class Command(enum.IntEnum):
+    """
+    :class:`~odoo.fields.One2many` and :class:`~odoo.fields.Many2many` fields
+    expect a special command to manipulate the relation they implement.
+    Internally, each command is a tuple where the first element is a mandatory
+    integer that identifies the command, the two next elements are arguments
+    related to the command.
+
+    Please refer to the :meth:`~odoo.models.BaseModel.write` documentation for
+    full details on the behavior of each command.
+    """
+
+    CREATE = 0
+    UPDATE = 1
+    DELETE = 2
+    UNLINK = 3
+    LINK = 4
+    CLEAR = 5
+    SET = 6
+
+    @staticmethod
+    def create(values):
+        """
+        Create new records in the comodel using ``values``, link the created
+        records to ``self``.
+        """
+        return (Command.CREATE, 0, values)
+
+    @staticmethod
+    def update(id, values):
+        """ Write ``values`` on the related record. """
+        return (Command.UPDATE, id, values)
+
+    @staticmethod
+    def delete(id):
+        """
+        Remove the related record from the database and remove its relation
+        with ``self``.
+        """
+        return (Command.DELETE, id)
+
+    @staticmethod
+    def unlink(id):
+        """ Remove the relation between ``self`` and the related record. """
+        return (Command.UNLINK, id)
+
+    @staticmethod
+    def link(id):
+        """ Add a relation between ``self`` and the related record. """
+        return (Command.LINK, id)
+
+    @staticmethod
+    def clear():
+        """ Remove all records from the relation with ``self``. """
+        return (Command.CLEAR,)
+
+    @staticmethod
+    def set(ids):
+        """ Replace the current relations of ``self`` by the given ones. """
+        return (Command.SET, 0, ids)
+
+
 class _RelationalMulti(_Relational):
     """ Abstract class for relational fields *2many. """
 
     # Important: the cache contains the ids of all the records in the relation,
     # including inactive records.  Inactive records are filtered out by
     # convert_to_record(), depending on the context.
 
+    command = Command
```

Comment:
???

Reply:
but idea from a time we didn't had the shortcut via `__init__.py`, useless now, am removing it

Reply:
I see. It looked confusing because there's a big diff below dealing with an existing `command` local so this looked weird.

Maybe calling the alias `commands` would make more sense if we keep it? Though as you say it might not be super useful if we're providing a convenient way to import it anyway.